examples/webhook.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/xanzy/go-gitlab"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// parse errors
+var (
+	errInvalidHTTPMethod             = errors.New("invalid HTTP Method")
+	errMissingGitLabEventHeader      = errors.New("missing X-Gitlab-Event Header")
+	errGitLabTokenVerificationFailed = errors.New("X-Gitlab-Token validation failed")
+	errEventNotFound                 = errors.New("event not defined to be parsed")
+	errParsingPayload                = errors.New("error parsing payload")
+)
+
+// webhookExample shows how to create a Webhook server to parse Gitlab events
+// listens /webhook endpoint on host 0.0.0.0 port 8080
+func webhookExample() {
+	wh := webhook{
+		Secret: "your-gitlab-secret",
+		// only listen for MergeRequest and Pipeline events
+		EventsToAccept: []gitlab.EventType{gitlab.EventTypeMergeRequest, gitlab.EventTypePipeline},
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/webhook", wh)
+	_ = http.ListenAndServe("0.0.0.0:8080", mux)
+}
+
+// webhook is a HTTP Handler for Gitlab Webhook events
+type webhook struct {
+	Secret         string
+	EventsToAccept []gitlab.EventType
+}
```

Comment:
Please place structs in the beginning of the file (for consistency). So just below the import section in this case...